src/main/kotlin/com/hypto/iam/server/apis/OrganizationApi.kt

```diff
@@ -28,17 +33,27 @@ import org.koin.ktor.ext.inject
  * API to create organization in IAM.
  */
 fun Route.createOrganizationApi() {
-    val service: OrganizationsService by inject()
+    val organizationService: OrganizationsService by inject()
+    val passcodeRepo: PasscodeRepo by inject()
+    val appConfig: AppConfig by inject()
     val gson: Gson by inject()
 
     post("/organizations") {
-        val request = call.receive<CreateOrganizationRequest>().validate()
-        val passcodeStr = call.principal<ApiPrincipal>()?.tokenCredential?.value
-        val (organization, tokenResponse) = service.createOrganization(
-            request.name,
-            description = request.description ?: "",
-            rootUser = request.rootUser,
-            passcodeStr = passcodeStr
+        val secretKey = Constants.SECRET_PREFIX + appConfig.app.secretKey
+        val passcodeStr = call.principal<ApiPrincipal>()?.tokenCredential?.value!!
+
+        // If secretKey is provided, then fetch use request body else use metadata from passcode table
```

Comment:
IAM should support different sign-up strategies.
1. User already entered admin details before verification.
2. User enters admin details after email verification.
3. Partial sign-ups where user enters credentials but email is not verified.

Logic here can be to
1. Check passcode table or sign-up request object and if the admin details are available, create org
2. If data is available in both the places or if the data is available in no place, throw 400.
3. For partial sign-ups support, we can give temporary token to play around in sandbox. (This can be added later)

Reply:
Added support for 1 & 2